### PR TITLE
Properly closing grpc connections on deregister

### PR DESCRIPTION
**Pull request by andrehp**
The gRPC RPC client is currently leaking connections because they are not being properly closed.

References:
https://github.com/grpc/grpc-go/issues/2086
https://godoc.org/google.golang.org/grpc#ClientConn.NewStream

### CODE REVIEW

File: cluster/grpc_rpc_client.go [hunk 1]
```diff
@@ -75,6 +75,11 @@ func NewGRPCClient(
 	return gs, nil
 }
 
+type grpcServer struct {
```
Comment (felipejfc) [inline cluster/grpc_rpc_client.go]: nit: doesn't make sense to call this struct grpcClient? and also the "srv" attribute is actually a client, sounds kinda strange...

File: cluster/grpc_rpc_client.go [hunk 2]
```diff
@@ -148,14 +162,14 @@ func (gs *GRPCClient) BroadcastSessionBind(uid string) error {
 	}
 	fid, _ := gs.bindingStorage.GetUserFrontendID(uid, gs.server.Type)
 	if fid != "" {
-		if c, ok := gs.clientMap.Load(fid); ok {
+		if s, ok := gs.clientMap.Load(fid); ok {
```
Comment (felipejfc) [inline cluster/grpc_rpc_client.go]: c? :p

File: cluster/grpc_rpc_client.go (hunk 1 above)
Comment (andrehp) [reply cluster/grpc_rpc_client.go]: I thought about it, my line of thought was that this struct contained information about a grpc server. But thinking now I tend to agree with you, I will change grpcServer to grpcClient and srv to cli

File: cluster/grpc_rpc_client.go (hunk 2 above)
Comment (andrehp) [reply cluster/grpc_rpc_client.go]: Gonna change that as well

File: cluster/grpc_rpc_client.go (hunk 1 above)
Comment (Henrod) [reply cluster/grpc_rpc_client.go]: We already have `GRPCClient` in this file. I would recommend the name `grpcConn`

**Pull request merged**